validator/src/main/java/com/amazon/aoc/clients/PrometheusClient.java

```diff
@@ -0,0 +1,97 @@
+package com.amazon.aoc.clients;
+
+import com.amazon.aoc.models.Context;
+import com.amazon.aoc.models.prometheus.PrometheusMetric;
+import com.amazon.aoc.models.prometheus.PrometheusQueryResult;
+import com.amazonaws.auth.AWS4Signer;
+import com.amazonaws.auth.profile.ProfileCredentialsProvider;
+import com.amazonaws.http.AWSRequestSigningApacheInterceptor;
+import com.amazonaws.util.IOUtils;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.extern.log4j.Log4j2;
+import okhttp3.HttpUrl;
+import org.apache.http.HttpResponse;
+import org.apache.http.client.HttpClient;
+import org.apache.http.client.config.RequestConfig;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.utils.URIBuilder;
+import org.apache.http.impl.client.DefaultHttpRequestRetryHandler;
+import org.apache.http.impl.client.HttpClients;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.List;
+import java.util.Objects;
+
+@Log4j2
+public class PrometheusClient {
```

Comment:
i'm assuming this is to call the sample app? any possibility we can make it generic? name it as "PullModeSampleAppClient"?

Reply:
Oh, this one is to call our Cortex Instance. The ExpectedResultClient is actually the one that calls the sample app. I'll rename both clients to make it more clear.